independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/LazyValue.java

```diff
@@ -34,9 +42,9 @@ public T getIfPresent() {
     }
 
     public void clear() {
-        synchronized (this) {
```

Comment:
you can check outside of this critical zone if it has already happened and save entering into it.
@mkouba do we expect to be able to init it again, after a clear?